superduperdb/base/datalayer.py

```diff
@@ -373,6 +347,23 @@ def predict(
             return Document(out), [Document(x) for x in context]
         return Document(out), []
 
+    def validate(self, query: ExecuteQuery, *args, **kwargs) -> None:
+        """
+        Validate a query on the db.
+
+        :param query: select, insert, delete, update,
+        """
+        if not isinstance(query, (Delete, Select, Update)):
+            return
+        assert isinstance(query, (Delete, Select, Update))
+        try:
+            if query.variables:
+                query.set_variables(self).validate(self)
+            else:
+                query.validate(self)
+        except exceptions.QueryValidationError:
+            raise
```

Comment:
It makes no sense to catch an exception just to immediately raise it again, so this can be simplified a bit.
```suggestion
        if query.variables:
            query = query.set_variables(self)
        query.validate(self)
```

Reply:
Agreed

Reply:
@Jamim ah this was intended for debugging, had put a breakpoint

Reply:
Im removing this one